sdks/python/apache_beam/runners/portability/fn_api_runner_test.py

```diff
@@ -543,6 +545,15 @@ def process(
       self.assertEqual(1, len(counters))
       self.assertEqual(counters[0].committed, len(''.join(data)))
 
+  def test_deduplication_by_id(self):
+    with self.create_pipeline() as p:
+      res = (p
+             | beam.Create([('id_1', 'value_1'),
+                            ('id_2', 'value_2'),
+                            ('id_1', 'value_1')])
+             | DeduplicationByUniqueId())
```

Comment:
Perhaps call this "DeduplicateByKey"?

Reply:
I would call it `ChoosePerKey` or something like that since what it actually does is choose an arbitrary element for each key. Test cases should include different values for the same key (will require a flexible result matching)

Reply:
I think there is an assumption of using this transform: under any situation, one unique id should map one specific value, for any 2 same unique id, the value should also be the same. Thus we can deduplicate the value by this id. One typical usage is reading from certain message queue, any message is paired with one unified ID.

Reply:
I understand. It does allow inconsistency but not too much risk since anyone will immediately see that a key might have two values and they have to be careful. FWIW in Java is it arranged slightly differently into [`Distinct.withRepresentativeValues`](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Distinct.java#L81). But that way is not as cross-language-friendly as precomputing the keys.